tests/foreman/cli/test_docker.py

```diff
@@ -1343,6 +1343,128 @@ def test_positive_pull_image(self):
+            'id': content_view['versions'][0]['id'],
+            'to-lifecycle-environment-id': lce['id'],
+        })
+        LifecycleEnvironment.update({
+            'registry-name-pattern': registry_name_pattern,
+            'registry-unauthenticated-pull': 'false',
+            'id': lce['id'],
+            'organization-id': self.org['id'],
+        })
+        expected_pattern = "{}-{}/{}".format(pattern_prefix,
+                                             content_view['label'],
+                                             docker_upstream_name).lower()
+
+        # 3. Try to search for docker images on Satellite
+        remote_search_command = 'docker search {0}/{1}'.format(
+                settings.server.hostname,
+                docker_upstream_name)
+        result = ssh.command(remote_search_command,
+                             hostname=self.docker_host.ip_addr)
+        self.assertEqual(result.return_code, 0)
+        self.assertLessEqual(len(result.stdout), 2)
+
+        # 4. Use Docker client to login to Satellite docker hub
+        result = ssh.command(
+                'docker login -u {} -p {} {}'.format(
+                        settings.server.admin_username,
+                        settings.server.admin_password,
+                        settings.server.hostname),
+                hostname=self.docker_host.ip_addr
+        )
+        self.assertEqual(result.return_code, 0)
+
+        # 5. Search for docker images
+        result = ssh.command(remote_search_command,
+                             hostname=self.docker_host.ip_addr)
+        self.assertEqual(result.return_code, 0)
+        self.assertGreater(len(result.stdout), 2)
+        print("{}".format(result.stdout))
+        self.assertIn("{}/{}".format(settings.server.hostname,
+                                     expected_pattern),
+                      "\n".join(result.stdout))
+
+        # 6. Use Docker client to log out of Satellite docker hub
+        result = ssh.command(
+                'docker logout {}'.format(settings.server.hostname),
+                hostname=self.docker_host.ip_addr
+        )
+        self.assertEqual(result.return_code, 0)
+
+        # 7. Set "Unauthenticated Pull" option to true
+        LifecycleEnvironment.update({
+            'registry-unauthenticated-pull': 'true',
+            'id': lce['id'],
+            'organization-id': self.org['id'],
+        })
+
+        # 8. Search for docker images
+        result = ssh.command(remote_search_command,
+                             hostname=self.docker_host.ip_addr)
+        self.assertEqual(result.return_code, 0)
+        self.assertGreater(len(result.stdout), 2)
+        self.assertIn("{}/{}".format(settings.server.hostname,
+                                     expected_pattern),
+                      "\n".join(result.stdout))
+
+        # cleanup: remove lce so test can be re-run
+        LifecycleEnvironment.delete({'id': lce['id']})
```

Comment:
that is not a cleanup, because in case of any failure, you will never get to that point. Maybe, you can look and use something from:
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/cleanup.py

if existing lce can really block further runs, please check usage of mentioned idea in our code